### PR TITLE
[WIP] `No Cure!` modification

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
`No Cure!` is `LUA` mod which causes instant death in 72 hours after being `infected`.

There is `No Cure!`, so even if you clean the wound and treat the infection - you will still die 72 hours after you've received the infection.

N.B.

You won't die if you somehow manage to remove the `infected` effect before `on_minute_passed` callback is triggered

default

default

default

default

### CODE REVIEW

**Comment by Mecares** [comment]
Could you make it so that certain items that normally would cure the infection instead delay the inevitable? Like taking antibiotics delay it for say a few hours? Being dead after 24 hours no matter what you do does not seem much fun.

**Comment by Night-Pryanik** [comment]
> Could you make it so that certain items that normally would cure the infection instead delay the inevitable? Like taking antibiotics delay it for say a few hours?

I'd say there should be either antivirals, as "traditional" zombification reason is a virus, not a bacteria, or it should be limited to royal jelly only. IIRC lore-wise everyone is contaminated with goo, player and npcs just have more potent immunity system, so they hadn't die as everyone else. Goo became more lethal in zeds' bodies (?), so survivors can't cope with modified infection by themselves.

**Comment by tinukedaya** [comment]
Maybe make it longer than 24 hours? If it's inevitable, give the player time to do something to go off with a style! 

And, throwing in some other ideas to make the wait for death actually worth playing and interesting, after you are infected: 
- As you're turning into a zombie, you do not need to sleep at all after the infection.
- Stats will start to change... STR goes up, INT goes down.. .not sure about the others. 
- Based on where the bite is, you will start getting mutations. Being bitten in the legs will start to give you leg mutations after some hours... After some time you might just get random mutations every few hours. First slow, just like drinking mutagen, later like drinking serums.

File: data/mods/No_Cure/main.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+function get_death_turn()
+
+  return tonumber(player:get_value("No_Cure_DeathTurn"))
+
+end
+
+function set_death_turn(death_turn)
+
+  player:set_value("No_Cure_DeathTurn", tostring(death_turn))
+  if get_death_turn() ~= nil then
```
Comment (BevapDin) [inline data/mods/No_Cure/main.lua]: You have just set the value, how could the getter ever return `nil`?

Comment (ZhilkinSerg) [reply data/mods/No_Cure/main.lua]: Thanks! That message should've move elsewhere.

**Comment by ZhilkinSerg** [comment]
> Maybe make it longer than 24 hours? If it's inevitable, give the player time to do something to go off with a style!

Okay - 3 days and not a minute more.

**Comment by ituluwituluwzev** [comment]
If it is exactly 72 hours, there could be some interesting flavor text that informs the player the end is neigh, when it is 71 hour 50 minute.

**Comment by ZhilkinSerg** [comment]
So, there are a lot of different ideas about zombie infection. I guess I need to sleep over this.

**Comment by secretfire42** [comment]
Here's a silly idea: Why not make it so that taking MUTAGEN slows it down - and passing a threshold cures you? Would finally make an in-universe reason (other then JUST BECAUSE) to take mutagen. You could even add some notes from some scientists who survived in isolated portions of labs somewhere (for awhile) noting it.

**Pull request closed without merging**